webgpu.h

```diff
@@ -845,8 +864,7 @@ typedef void (*WGPUProcAdapterRequestDevice)(WGPUAdapter adapter, WGPUDeviceDesc
 typedef void (*WGPUProcBufferDestroy)(WGPUBuffer buffer);
 typedef void const * (*WGPUProcBufferGetConstMappedRange)(WGPUBuffer buffer);
 typedef void * (*WGPUProcBufferGetMappedRange)(WGPUBuffer buffer);
-typedef void (*WGPUProcBufferMapReadAsync)(WGPUBuffer buffer, WGPUBufferMapReadCallback callback, void * userdata);
-typedef void (*WGPUProcBufferMapWriteAsync)(WGPUBuffer buffer, WGPUBufferMapWriteCallback callback, void * userdata);
+typedef void (*WGPUProcBufferMapAsync)(WGPUBuffer buffer, WGPUMappingFlagFlags flags, size_t offset, size_t size, WGPUBufferMapWriteCallback callback, void * userdata);
```

Comment:
I believe `offset` should be `uint64_t`. Actually now that I think about it, that's a pretty good reason for both read and write to have the `offset` and `size` args. If the CPU can't address all of the GPU memory then it would be impossible to map the whole buffer, and then `getMappedRange()` at the start of it, and at the end of it.

Reply:
nobody should operate on staging buffers that can't be addressed on CPU in full, that would be quite crazy :)